python/uv/private/pin.bzl

```diff
@@ -0,0 +1,184 @@
+# Copyright 2024 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A simple macro to pin the requirements.
+"""
+
+load("@bazel_skylib//rules:write_file.bzl", "write_file")
+load("//python:py_binary.bzl", "py_binary")
+load("//python/private:bzlmod_enabled.bzl", "BZLMOD_ENABLED")  # buildifier: disable=bzl-visibility
+
+_REQUIREMENTS_TARGET_COMPATIBLE_WITH = select({
```

Comment:
Does uv only provide linux/mac binaries...?

Reply:
No, `windows` binaries are also provided, just the `genrule` implementation is in shell.